packages/server/lib/cloud/studio/StudioLifecycleManager.ts

```diff
@@ -342,18 +346,45 @@ export class StudioLifecycleManager {
     if (this.studioManager) {
       debug('Studio ready - calling listener immediately')
       listener(this.studioManager)
-
-      // If the studio bundle is local, we need to register the listener
-      // so that we can reload the studio when the bundle changes
-      if (process.env.CYPRESS_LOCAL_STUDIO_PATH) {
-        this.listeners.push(listener)
-      }
+      this.listeners.push(listener)
     } else {
       debug('Studio not ready - registering studio ready listener')
       this.listeners.push(listener)
     }
   }
 
+  public getCurrentStatus (): StudioStatus | undefined {
+    return this.lastStatus
+  }
+
+  public retry (): void {
+    if (!this.ctx) {
+      debug('No ctx available, cannot retry studio initialization')
+
+      return
+    }
+
+    debug('Retrying studio initialization')
+
+    this.studioManager = undefined
+    this.studioManagerPromise = undefined
+    this.lastStatus = undefined
+
+    // Clear any cached failed bundle downloads to allow for retrying
+    const cacheSize = StudioLifecycleManager.hashLoadingMap.size
```

Comment:
Can't we just clear out the failed request rather than all of them? Maybe retry could take the bundle hash?